nano/node/messages.cpp

```diff
@@ -796,6 +796,11 @@ bool nano::publish::operator== (nano::publish const & other_a) const
 	return *block == *other_a.block;
 }
 
+std::string nano::publish::to_string () const
+{
+	return block->to_json ();
```

Comment:
It should be printing the header too.
The other to_string functions print the header too.